src/service/resources-impl.js

```diff
@@ -1037,11 +1038,16 @@ export class Resources {
    * @private
    */
   calcTaskTimeout_(task) {
+    const now = timer.now();
+
     if (this.exec_.getSize() == 0) {
-      return 0;
+      // Scale off the first visible time, so penalized tasks must wait a
+      // second or two to run. After we have been visible for a time, we no
+      // longer have to wait.
+      const penalty = task.priority * PRIORITY_PENALTY_TIME_;
+      return Math.max(penalty - (now - this.firstVisibleTime_), 0);
```

Comment:
That's intentional. I couldn't make up my mind what to do here, and decided to do the same thing as we were currently (returning 0). I'd be fine with returning `penalty` here without the subtraction, though.